src/context/layers/impact.ts

```diff
@@ -168,76 +183,134 @@ export async function fetchImpactLayerData(
     );
   }
 
-  // Calculate a bounding box for each feature that we have baseline data for
-  const matchingFeatures = adminBoundaries.features.reduce(
-    (acc, f) => {
-      const feature = f as GeoJsonBoundary;
-      const baseline =
-        featureIntersectsImage(feature, image) &&
-        getBaselineDataForFeature(feature, baselineData);
-      return baseline
-        ? acc.concat({
-            id: baseline.adminKey,
-            baseline,
-            feature,
-            bounds: bbox(feature),
-          })
-        : acc;
-    },
-    [] as {
-      id: string;
-      baseline: BaselineRecord;
-      feature: GeoJsonBoundary;
-      bounds: ReturnType<typeof bbox>;
-    }[],
-  );
+  // eslint-disable-next-line fp/no-mutation
+  let activeFeatures: GeoJsonBoundary[];
+  if (layer.api) {
+    const wcsUrl = getWCSLayerUrl({
+      layer: hazardLayerDef,
+      extent,
+      date,
+    } as LayerDataParams<WMSLayerProps>);
+    const apiUrl = layer.api.url;
+
+    const apiData = {
+      geotiff_url: wcsUrl,
+      zones_url: layer.api.zonesUrl,
+      group_by: layer.api.groupBy,
+      geojson_out: 'false',
+    };
 
-  // Loop over the features and grab pixels that are contained by the feature.
-  const buckets = matchingFeatures.reduce((acc, { id, feature }) => {
-    const contained = pixelsInFeature(
-      feature,
-      rasters[0],
-      rasters.width,
-      transform,
+    const aggregateData = await fetchApiData(apiUrl, apiData);
+
+    const mergedFeatures = mergeFeaturesByProperty(
+      baselineData.features.features,
+      aggregateData,
+      layer.api.groupBy,
+      operation,
     );
-    return contained.length > 0 ? { ...acc, [id]: contained } : acc;
-  }, {} as { [key: string]: number[] });
 
-  const operation = layer.operation || 'median';
+    // eslint-disable-next-line fp/no-mutation
+    activeFeatures = mergedFeatures.filter(feature => {
+      const scaled = scaleValueIfDefined(
+        feature.properties.stats_median,
+        scale,
+        offset,
+      );
+      return thresholdOrNaN(scaled, layer.threshold);
+    }) as GeoJsonBoundary[];
```

Comment:
You shouldn't need to cast this here - `mergeFeaturesByProperty` should have a typed return value.